raycicmd/rayci_pipeline.go

```diff
@@ -43,7 +43,7 @@ var (
 	commandStepAllowedKeys = []string{
 		"command", "commands", "priority", "parallelism", "if",
 		"label", "name", "key", "depends_on", "soft_fail", "matrix",
-		"allow_dependency_failure",
+		"allow_dependency_failure", "concurrency", "concurrency_group",
```

Comment:
can you maybe follow up with an `allowed_concurrency_group` in the config as an allowlist of `concurrency_group` names that can be used?

Reply:
why do we want to restrict the usage of these keys through a flag?

Reply:
I do not want untrusted code pipelines being able to take concurrency group quotas from the ones that are allocated / used to trusted pipelines.

Reply:
what actions we want to prevent the untrusted pipelines to do that the limited set of `concurrency_group` name can help with?

Reply:
.. vs the cost of users who have been using these fields in buildkite and do not know why they do not work in rayci

Reply:
synced offline.

another idea can be auto adding a prefix to the `concurrency_group` name (and the default prefix is empty). this will make it possible that different pipelines are sharded into different namespaces. like `premerge-` , `postmerge-`, `releasetest-`